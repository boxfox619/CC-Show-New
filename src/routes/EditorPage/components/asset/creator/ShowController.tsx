import * as React from 'react';
import styled from 'styled-components';
import GradientButtonItem from './GradientButtonItem';
import Profile from './Profile';
import PointModel from 'src/core/models/PointModel';

const Container = styled.div`
    background-color: white;
    box-shadow: 0 1px 7px 0 rgba(0, 0, 0, 0.26);
    height: 100%;
    display: inline-block;
`

const ButtonGroup = styled.div`
    padding: 20px;
    display: flex;
    flex-flow: column;
    width: 200px;
    & > * {
        margin-bottom: 5px;
    }
`

const SplitBar = styled.div`
    display: block;
    height: 1.5px;
    background-color: rgba(33,33,33,0.1);
    width: 40px;
    margin-left: 20px;
`
interface OwnProps {
    name: string,
    email: string,
    thumbnail: string,
    addAsset: (type: string, point: PointModel) => void,
    toggleAssetManager: () => void,
    toggleSlideManager: () => void,
    toggleSlideShow: () => void
}

type Props = OwnProps & React.HTMLAttributes<HTMLDivElement>;

const ShowController: React.FC<Props> = (props: Props) => {
    const divProps = props as React.HTMLAttributes<HTMLDivElement>;
    const createAssetByType = (assetType: string) => () => props.addAsset(assetType, { x: 0, y: 0 });
    return (
        <Container {...divProps}>
            <Profile
                thumbnail={props.thumbnail}
                name={props.name}
                subName={props.email} />
            <SplitBar />
            <ButtonGroup>
                <GradientButtonItem label="텍스트" onClick={createAssetByType('text')}/>
                <GradientButtonItem label="비디오" onClick={createAssetByType('video')}/>
                <GradientButtonItem label="이미지" onClick={createAssetByType('image')}/>
                <GradientButtonItem label="도형" onClick={createAssetByType('shape')}/>
                <GradientButtonItem label="기타" onClick={props.toggleAssetManager}/>
            </ButtonGroup>
            <SplitBar />
            <ButtonGroup>
                <GradientButtonItem label="슬라이드 리스트" onClick={props.toggleSlideManager}/>
                <GradientButtonItem label="슬라이드 쇼" onClick={props.toggleSlideShow}/>
            </ButtonGroup>
        </Container>
    )
}

export default ShowController;