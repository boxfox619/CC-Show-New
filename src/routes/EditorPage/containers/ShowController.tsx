import * as React from 'react';
import styled from 'styled-components';
import { CreateAssetPayload } from '../models/payload';
import { AssetType } from '../../../models';
import GradientButtonItem from '../components/GradientButtonItem';
import Profile from '../components/Profile';
import { ShapeArrow1 } from '../../../core/assets';

const Container = styled.div`
    background-color: white;
    box-shadow: 0 1px 7px 0 rgba(0, 0, 0, 0.26);
    height: 100%;
    display: inline-block;
`

const ButtonGroup = styled.div`
    padding: 20px;
    display: flex;
    flex-flow: column;
    width: 200px;
    & > * {
        margin-bottom: 5px;
    }
`

const SplitBar = styled.div`
    display: block;
    height: 1.5px;
    background-color: rgba(33,33,33,0.1);
    width: 40px;
    margin-left: 20px;
`
interface OwnProps {
    name: string,
    email: string,
    thumbnail: string,
    visibleSlideManager: boolean,
    addAsset: (payload: CreateAssetPayload) => void,
    toggleAssetManager: () => void,
    toggleSlideManager: () => void,
    toggleSlideShow: () => void
}

type Props = OwnProps & React.HTMLAttributes<HTMLDivElement>;

const ShowController: React.FC<Props> = (props: Props) => {
    const divProps = props as React.HTMLAttributes<HTMLDivElement>;
    const createAssetByType = (assetType: AssetType, value?: any) => () => props.addAsset(new CreateAssetPayload(assetType, { x: 0, y: 0 }, value));
    return (
        <Container {...divProps}>
            <Profile
                thumbnail={props.thumbnail}
                name={props.name}
                subName={props.email} />
            <SplitBar />
            <ButtonGroup>
                <GradientButtonItem label="텍스트" onClick={createAssetByType(AssetType.Text, 'Welcome to CC-Show!')} />
                <GradientButtonItem label="비디오" onClick={createAssetByType(AssetType.Video)} />
                <GradientButtonItem label="이미지" onClick={createAssetByType(AssetType.Image, 'http://poooo.ml/data/editor/1810/aa7462a202b36ecf40db2f8e44d4f594_1539011087_018.gif')} />
                <GradientButtonItem label="도형" onClick={createAssetByType(AssetType.Shape, ShapeArrow1)} />
                <GradientButtonItem label="기타" onClick={props.toggleAssetManager} />
            </ButtonGroup>
            <SplitBar />
            <ButtonGroup>
                <GradientButtonItem label="슬라이드 리스트" onClick={props.toggleSlideManager} active={props.visibleSlideManager} />
                <GradientButtonItem label="슬라이드 쇼" onClick={props.toggleSlideShow} />
            </ButtonGroup>
        </Container>
    )
}

export default ShowController;