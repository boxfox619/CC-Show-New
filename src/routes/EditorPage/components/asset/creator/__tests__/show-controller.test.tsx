import * as React from 'react';
import * as Enzyme from 'enzyme';
import ShowController from '../ShowController';

describe('<ShowController />', () => {
    const mockFn = jest.fn();
  it('should render one <div> & div label', () => {
    const wrapper = Enzyme.shallow(<ShowController addAsset={new mockFn()}/>);
    expect(wrapper).toMatchSnapshot();
  });
});