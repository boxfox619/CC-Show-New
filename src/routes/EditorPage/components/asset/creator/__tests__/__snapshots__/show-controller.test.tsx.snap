// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowController /> should render one <div> & div label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowController
    addAsset={mockConstructor {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "addAsset": mockConstructor {},
      "children": Array [
        <Profile
          name="홍길동"
          subName="치킨비어"
          thumbnail="https://avatars1.githubusercontent.com/u/14067209?s=460&v=4"
        />,
        <ForwardRef />,
        <ForwardRef>
          <GradientButtonItem
            label="텍스트"
            onClick={[Function]}
          />
          <GradientButtonItem
            label="비디오"
            onClick={[Function]}
          />
          <GradientButtonItem
            label="이미지"
            onClick={[Function]}
          />
          <GradientButtonItem
            label="도형"
            onClick={[Function]}
          />
          <GradientButtonItem
            label="기타"
          />
        </ForwardRef>,
        <ForwardRef />,
        <ForwardRef>
          <GradientButtonItem
            label="슬라이드 리스트"
          />
          <GradientButtonItem
            label="슬라이드 쇼"
          />
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "홍길동",
          "subName": "치킨비어",
          "thumbnail": "https://avatars1.githubusercontent.com/u/14067209?s=460&v=4",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
    display: block;
    height: 1.5px;
    background-color: rgba(33,33,33,0.1);
    width: 40px;
    margin-left: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GradientButtonItem
              label="텍스트"
              onClick={[Function]}
            />,
            <GradientButtonItem
              label="비디오"
              onClick={[Function]}
            />,
            <GradientButtonItem
              label="이미지"
              onClick={[Function]}
            />,
            <GradientButtonItem
              label="도형"
              onClick={[Function]}
            />,
            <GradientButtonItem
              label="기타"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "텍스트",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "비디오",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "이미지",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "도형",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "기타",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "rules": Array [
              "
    padding: 20px;
    display: flex;
    flex-flow: column;
    width: 200px;
    & > * {
        margin-bottom: 5px;
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
    display: block;
    height: 1.5px;
    background-color: rgba(33,33,33,0.1);
    width: 40px;
    margin-left: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GradientButtonItem
              label="슬라이드 리스트"
            />,
            <GradientButtonItem
              label="슬라이드 쇼"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "슬라이드 리스트",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "label": "슬라이드 쇼",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "rules": Array [
              "
    padding: 20px;
    display: flex;
    flex-flow: column;
    width: 200px;
    & > * {
        margin-bottom: 5px;
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": true,
        "rules": Array [
          "
    background-color: white;
    box-shadow: 0 1px 7px 0 rgba(0, 0, 0, 0.26);
    height: 100%;
    display: inline-block;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "addAsset": mockConstructor {},
        "children": Array [
          <Profile
            name="홍길동"
            subName="치킨비어"
            thumbnail="https://avatars1.githubusercontent.com/u/14067209?s=460&v=4"
          />,
          <ForwardRef />,
          <ForwardRef>
            <GradientButtonItem
              label="텍스트"
              onClick={[Function]}
            />
            <GradientButtonItem
              label="비디오"
              onClick={[Function]}
            />
            <GradientButtonItem
              label="이미지"
              onClick={[Function]}
            />
            <GradientButtonItem
              label="도형"
              onClick={[Function]}
            />
            <GradientButtonItem
              label="기타"
            />
          </ForwardRef>,
          <ForwardRef />,
          <ForwardRef>
            <GradientButtonItem
              label="슬라이드 리스트"
            />
            <GradientButtonItem
              label="슬라이드 쇼"
            />
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "홍길동",
            "subName": "치킨비어",
            "thumbnail": "https://avatars1.githubusercontent.com/u/14067209?s=460&v=4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
    display: block;
    height: 1.5px;
    background-color: rgba(33,33,33,0.1);
    width: 40px;
    margin-left: 20px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GradientButtonItem
                label="텍스트"
                onClick={[Function]}
              />,
              <GradientButtonItem
                label="비디오"
                onClick={[Function]}
              />,
              <GradientButtonItem
                label="이미지"
                onClick={[Function]}
              />,
              <GradientButtonItem
                label="도형"
                onClick={[Function]}
              />,
              <GradientButtonItem
                label="기타"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "텍스트",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "비디오",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "이미지",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "도형",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "기타",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bZQynM",
              "isStatic": true,
              "rules": Array [
                "
    padding: 20px;
    display: flex;
    flex-flow: column;
    width: 200px;
    & > * {
        margin-bottom: 5px;
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bZQynM",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
    display: block;
    height: 1.5px;
    background-color: rgba(33,33,33,0.1);
    width: 40px;
    margin-left: 20px;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GradientButtonItem
                label="슬라이드 리스트"
              />,
              <GradientButtonItem
                label="슬라이드 쇼"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "슬라이드 리스트",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "label": "슬라이드 쇼",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bZQynM",
              "isStatic": true,
              "rules": Array [
                "
    padding: 20px;
    display: flex;
    flex-flow: column;
    width: 200px;
    & > * {
        margin-bottom: 5px;
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bZQynM",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": true,
          "rules": Array [
            "
    background-color: white;
    box-shadow: 0 1px 7px 0 rgba(0, 0, 0, 0.26);
    height: 100%;
    display: inline-block;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
